learnhubmk/app
# [FE] Implement Role-Based Middleware for Page Authorization [444]

## Authentication Refactoring

### Changes
- Migrated from custom localStorage-based authentication to NextAuth
- Removed unnecessary authentication context (`authContext.tsx`)
- Deleted obsolete authentication API files (`authApi.ts`, `session.ts`)
- Updated axios instance to use NextAuth session for token management
- Created new login and logout hooks in queries folder
- Added basic unauthorized page
- Implemented middleware to protect routes

### Key Improvements
- More secure authentication method
- Simplified authentication flow
- Better integration with Next.js authentication patterns
- Removed client-side token management

### Breaking Changes
- Removed custom authentication context
- Changed login and logout mechanisms
- Requires updates to environment variables (added `AUTH_SECRET`, `NEXTAUTH_URL`)

### Testing instructions
-- Go to route /content-panel/tags (a protected route)
- you should not be allowed and redirected to content-panel/login

-- Login using content-manager account at /content-panel/login
- try accessing the protected route /content-panel/tags and modify tag. It should be successful.
- try accessing /admin-panel/  -> you will land on unauthorized page
